admin-dev/themes/new-theme/js/pages/product/edit/virtual-product-manager.ts

```diff
@@ -76,4 +77,33 @@ export default class VirtualProductManager {
   private showContent(): void {
     this.$fileContentContainer.removeClass('d-none');
   }
+
+  private fillVirtualProductNameField(): void {
+    $(ProductMap.virtualProduct.fileUploadField).on('change', () => {
+      // do not fill the name field if it's not empty
+      if ($(ProductMap.virtualProduct.fileNameField).val() !== '') {
+        return;
+      }
+
+      let fullPath = $(ProductMap.virtualProduct.fileUploadField).val();
+
+      if (Array.isArray(fullPath) || typeof fullPath === 'undefined') {
+        throw new Error('Full path of the file is not supposed to be an array');
+      } else {
+        // Convert to string in any other case
+        fullPath = fullPath.toString();
+      }
+
+      alert(fullPath);
+
+      // get file name from full path
+      const startIndex = (fullPath.indexOf('\\') >= 0 ? fullPath.lastIndexOf('\\') : fullPath.lastIndexOf('/'));
+      let filename = fullPath.substring(startIndex);
+
+      if (filename.indexOf('\\') === 0 || filename.indexOf('/') === 0) {
+        filename = filename.substring(1);
+      }
```

Comment:
Last index + 1 = First index + 1
```suggestion
      const startIndex = ['\\', '/'].reduce((index, search) => 0 <= index ? index : fullPath.lastIndexOf(search), -1) + 1;
      const filename = fullPath.substring(startIndex);
```